Composer/packages/client/src/recoilModel/atoms/appState.ts

```diff
@@ -369,3 +369,33 @@ export const showGetStartedTeachingBubbleState = atom<boolean>({
   key: getFullyQualifiedKey('showGetStartedTeachingBubbleState'),
   default: false,
 });
+
+export const showAuthDialogState = atom<boolean>({
+  key: getFullyQualifiedKey('showAuthDialog'),
+  default: false,
+});
+
+export const primaryTokenState = atom<string>({
+  key: getFullyQualifiedKey('primaryToken'),
+  default: '',
+});
+
+export const graphTokenState = atom<string>({
+  key: getFullyQualifiedKey('graphToken'),
+  default: '',
+});
+
+export const availableTenantsState = atom<AzureTenant[]>({
+  key: getFullyQualifiedKey('availableTenants'),
+  default: [],
+});
+
+export const currentTenantState = atom<string>({
```

Comment:
consider currentTenantIdState as you are storing a string rather than the AzureTenant structure.  You could also have a selector to give you back the currentAzureTenant.

Reply:
done! good call.